OpenRA.Game/World.cs

```diff
@@ -189,7 +189,7 @@ internal World(ModData modData, Map map, OrderManager orderManager, WorldType ty
 			this.modData = modData;
 			Type = type;
 			OrderManager = orderManager;
-			Map = map;
+			Map = (IMap)map;
```

Comment:
Why not pass an IMap in constructor.

Reply:
There are two reasons. First `World.cs` is part of `OpenRA.Game`, so it is not mod code and therefore may need to use `Map.cs` without the interface. Secondly it does call some functions that rely on `Map` not `IMap`, such as `CacheModels`, which is out of scope to refactor at this stage. However since `World.cs` will likely always need to be in `OpenRA.Game`, there will always be core map functions that it can and will need to access that are not part of `IMap`.